File: jankenpo.rb
puts "===== Permainan Batu-Kertas-Gunting ====="
puts "Anggota Kelompok:"
puts "1. [Nama Anggota 1] - [NIM Anggota 1]"
puts "2. [Nama Anggota 2] - [NIM Anggota 2]"
puts "3. [Nama Anggota 3] - [NIM Anggota 3]"
puts ""

print "Masukkan nama Anda: "
nama_pengguna = gets.chomp

puts "\nPilihan Anda:"
puts "1: Batu"
puts "2: Kertas"
puts "3: Gunting"
puts ""

pilihan_pengguna = nil

until (1..3).include?(pilihan_pengguna)
  print "Masukkan pilihan (1-3): "
  pilihan_pengguna = gets.chomp.to_i

  unless (1..3).include?(pilihan_pengguna)
    puts "Pilihan tidak valid. Silakan pilih antara 1-3."
    puts ""
  end
end

pilihan_komputer = rand(1..3)

# Array of choices
choices = ["Batu", "Kertas", "Gunting"]

puts "\n"
puts "#{nama_pengguna} memilih #{choices[pilihan_pengguna - 1]}"
puts "Komputer memilih #{choices[pilihan_komputer - 1]}"
puts ""

if pilihan_pengguna == pilihan_komputer
  puts "Hasil: Seri!"
else
  case pilihan_pengguna
  when 1
    puts (pilihan_komputer == 3) ? "Hasil: #{nama_pengguna} Menang!" : "Hasil: #{nama_pengguna} Kalah!"
  when 2
    puts (pilihan_komputer == 1) ? "Hasil: #{nama_pengguna} Menang!" : "Hasil: #{nama_pengguna} Kalah!"
  when 3
    puts (pilihan_komputer == 2) ? "Hasil: #{nama_pengguna} Menang!" : "Hasil: #{nama_pengguna} Kalah!"
  end
end

File: jankenpo.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

func main() {
	fmt.Println("Game DKP - Batu Kertas Gunting")

	fmt.Println(" ")

	fmt.Println("Nama NIM")
	fmt.Println("Nama NIM")
	fmt.Println("Nama NIM")

	fmt.Println(" ")

	fmt.Println("Masukkan nama pemain: ")
	var namaPemain string
	fmt.Scanln(&namaPemain)
	fmt.Println(" ")
	fmt.Println("Pilih 1 (Batu), Pilih 2 (Kertas), atau Pilih 3 (Gunting)")
	var pilihanPemain int
	fmt.Scanln(&pilihanPemain)
	fmt.Println(" ")

	rand.Seed(time.Now().UnixNano())
	pilihanKomputer := rand.Intn(3) + 1

	var pilihanPemainStr string

	switch pilihanPemain {
	case 1:
		pilihanPemainStr = "Batu"
	case 2:
		pilihanPemainStr = "Kertas"
	case 3:
		pilihanPemainStr = "Gunting"
	default:
		fmt.Println("Pilihan tidak valid")
		return
	}

	var pilihanKomputerStr string

	switch pilihanKomputer {
	case 1:
		pilihanKomputerStr = "Batu"
	case 2:
		pilihanKomputerStr = "Kertas"
	case 3:
		pilihanKomputerStr = "Gunting"
	}

	fmt.Printf("Pilihan Anda: %s\n", pilihanPemainStr)
	fmt.Printf("Pilihan Komputer: %s\n", pilihanKomputerStr)
	fmt.Println(" ")

	switch {
	case pilihanPemain == pilihanKomputer:
		fmt.Println("Seri")
	case pilihanPemain == 1 && pilihanKomputer == 3:
		fmt.Println(namaPemain, "Menang!")
	case pilihanPemain == 2 && pilihanKomputer == 1:
		fmt.Println(namaPemain, "Menang!")
	case pilihanPemain == 3 && pilihanKomputer == 2:
		fmt.Println(namaPemain, "Menang!")
	default:
		fmt.Println("Komputer Menang!")
	}
}
